src/frequenz/client/reporting/_client.py

```diff
@@ -248,71 +224,33 @@ def dt2ts(dt: datetime) -> PBTimestamp:
             end=dt2ts(end_dt),
         )
 
-        list_filter = PBListMicrogridComponentsDataRequest.ListFilter(
+        list_filter = PBReceiveMicrogridComponentsDataStreamRequest.StreamFilter(
             time_filter=time_filter,
             resampling_options=PBResamplingOptions(resolution=resolution),
         )
 
         metrics_pb = [metric.to_proto() for metric in metrics]
 
-        page_token = None
-
-        while True:
-            pagination_params = PBPaginationParams(
-                page_size=page_size, page_token=page_token
-            )
+        request = PBReceiveMicrogridComponentsDataStreamRequest(
+            microgrid_components=microgrid_components_pb,
+            metrics=metrics_pb,
+            filter=list_filter,
+        )
 
-            response = await self._fetch_page(
-                microgrid_components=microgrid_components_pb,
-                metrics=metrics_pb,
-                list_filter=list_filter,
-                pagination_params=pagination_params,
+        try:
+            stream = self._stub.ReceiveMicrogridComponentsDataStream(
+                request, metadata=self._metadata
             )
-            if not response or response.is_empty():
-                break
-
-            yield response
 
-            page_token = response.next_page_token
-            if not page_token:
-                break
+            async for response in stream:
```

Comment:
This `async` causes mypy to fail.

Reply:
The cast doesn't seem to solve the problem. I have added a type ignore to this line.